### PR TITLE
WIP: Feature/healthcheck signal

### PR DESCRIPTION
**Pull request by chrisz100**
Solves https://github.com/GoogleContainerTools/kaniko/issues/17

### CODE REVIEW

File: pkg/commands/commands.go [hunk 1]
```diff
@@ -68,6 +68,8 @@ func GetCommand(cmd instructions.Command, buildcontext string) (DockerCommand, e
 		return &ArgCommand{cmd: c}, nil
 	case *instructions.ShellCommand:
 		return &ShellCommand{cmd: c}, nil
+	case *instructions.HealthCheckCommand:
+		return &HealthCheckCommand{cmd: c}, nil
```
Comment (dlorenc) [inline pkg/commands/commands.go]: From the build error, it looks like you'll have to translate from the dockerfile HealthCheck type to the go-containerregistry one.

Comment (chrisz100) [reply pkg/commands/commands.go]: exactly. Just looking into how to do this the most efficiently. 

Just creating a v1.HealthConfig and passing all values would work, but I wouldn't really regard this as future proof or clean tbh. Any alternatives you could think of?

**Comment by chrisz100** [comment]
So... I don't really like this way of doing it but it works and there is no way to convert struct to struct more fail-safe/ autonomous that I am aware of.

**Comment by dlorenc** [comment]
> So... I don't really like this way of doing it but it works and there is no way to convert struct to struct more fail-safe/ autonomous that I am aware of.

Hmm, if the structs are actually equivalent you might be able to cast them: https://beta.golang.org/doc/go1.8#language

That would then error if the types don't match which would probably be a good thing.

**Comment by chrisz100** [comment]
This is the case as of now as well sadly: 
`config.Healthcheck = v1.HealthConfig(h.cmd.Health)`
```
pkg/commands/healthcheck.go:44:38: cannot convert h.cmd.Health (type *container.HealthConfig) to type "github.com/GoogleContainerTools/kaniko/vendor/github.com/google/go-containerregistry/v1".HealthConfig
make: *** [out/executor] Error 2
```

On go 1.10

**Comment by dlorenc** [comment]
Hmm, that might be because one is a pointer and the other isn't. Could you give it one more try by dereferencing the h.cmd.Health before converting?

**Comment by chrisz100** [comment]
Same thing: 
```
test := &h.cmd.Health
converted := v1.HealthConfig(test)

--> 
pkg/commands/healthcheck.go:45:30: cannot convert test (type **container.HealthConfig) to type "github.com/GoogleContainerTools/kaniko/vendor/github.com/google/go-containerregistry/v1".HealthConfig
make: *** [out/executor] Error 2
```

**Comment by dlorenc** [comment]
Nice!

**Pull request merged**